### PR TITLE
#331 @cardstack/data load service

### PR DESCRIPTION
**Pull request by habdelra**
Addresses issue #331 

This is a first pass thru a client implementation of the cardstack/data service which provides a mechanism to load all the related models when retrieving data for a particular card in a specific format. Note that I have separated out the ember addon boiler plate code so feel free to ignore that commit when reviewing. 

Still need to think thru a bunch of items:
- [x] don't reload a record that you have already loaded
- [x] parallelize record loads that are not related to one another
- [x] more tests around `defaultIncludes` vs `fieldsets` clashing

### CODE REVIEW

File: packages/models/cardstack/code-generator.js [hunk 1]
```diff
@@ -37,6 +37,7 @@ define('@cardstack/models/generated/{{modelName}}', ['exports', '@cardstack/mode
      value: true
    });
    exports.default = _model.default.extend({
+     type: "{{modelName}}",
```
Comment (ef4) [inline packages/models/cardstack/code-generator.js]: This unlocks a bunch of other simplifications in places where we're using cs-model-type or model.constructor.name.

File: packages/data/addon/services/cardstack-data.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+export default Service.extend({
+  store: inject(),
+
+  init() {
+    this._super();
+    this._loadedRecordsCache = {};
+    this._contentTypeCache = {};
+  },
+
+  async load(type, id, format) {
```
Comment (ef4) [inline packages/data/addon/services/cardstack-data.js]: This API will eventually also want `branch` as an argument.

**Comment by ef4** [comment]
This looks good to me.

File: packages/data/addon/services/cardstack-data.js (hunk 2 above)
Comment (habdelra) [reply packages/data/addon/services/cardstack-data.js]: I'll add that parameter in there now so it wont be a mess to deal with in the future

Comment (habdelra) [reply packages/data/addon/services/cardstack-data.js]: or well... when we perform `store.findRecord()` how do we incorporate `branch` into that request? or is that not currently supported?

Comment (habdelra) [reply packages/data/addon/services/cardstack-data.js]: Nevermind, found it: https://github.com/cardstack/cardstack/blob/master/packages/routing/addon/routes/cardstack/common.js#L32

**Comment by habdelra** [comment]
@ef4 I'm thinking we should also add this service in the @cardstack/router 's commonModelHook.

**Comment by habdelra** [comment]
or well I can do that in a subsequent PR. I'm seeing that this works really nicely in our own cardstack app.

**Pull request merged**